<launch>

  <arg name="mav_name" value="hmcl"/>
  <arg name="odom_topic" default="/mavros/local_position/odom"/>

<!-- Ros time align Start-->
  <param name="use_sim_time" value="true"/>
  <node name="clock_relay" pkg="topic_tools" type="relay" args="/airsim_node/clock /clock --wait-for-start"/>
<!-- Ros time align End-->

<!-- Static transforms Start -->
	<include file="$(find exploration_manager)/launch/static_transforms.launch"/>
<!-- Static transforms End -->

<!-- Camera Pose Publisher Start -->
  <node pkg="exploration_manager" type="camera_pose.py" name="camera_pose" output="screen">
    <remap from ="/odom" to="$(arg odom_topic)"/>
    <remap from ="/transformed_pose" to="/hmcl/camera_pose"/>
  </node>
<!-- Camera Pose Publisher End -->

<!-- Camera Depth RGB Publisher Start -->
  <node pkg="exploration_manager" type="camera_depth_publisher.py" name="camera_depth_publisher" output="screen"></node>
<!-- Camera Depth RGB Publisher End -->

<!-- Camera Depth iamge to Pointcloud Start -->
    <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>
    <node pkg="nodelet" type="nodelet" name="nodelet1"
        args="load depth_image_proc/point_cloud_xyzrgb nodelet_manager">
      <remap from="rgb/camera_info" to="/camera/rgb/camera_info"/>
      <remap from="rgb/image_rect_color" to="/camera/rgb/image_raw"/>
      <remap from="depth_registered/image_rect" to="/camera/depth/image_raw"/>
      <remap from="/depth_registered/points" to="/camera/depth/image_raw/points"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />
    <node pkg="nodelet" type="nodelet" name="voxel_grid" args="load pcl/VoxelGrid pcl_manager" output="screen">
      <remap from="~input" to="/camera/depth/image_raw/points" />
      <rosparam>
        filter_field_name: z
        filter_limit_min: 0.1
        filter_limit_max: 5.1
        filter_limit_negative: False
        leaf_size: 0.07
      </rosparam>
    </node>

<!-- Camera Depth iamge to Pointcloud End -->

  <!-- <include file="$(find exploration_manager)/launch/rviz.launch">
    <arg name="rviz_config" value="$(find exploration_manager)/config/gazebo.rviz"/>
  </include> -->


</launch>
